Jenkinsfile

```diff
@@ -1,135 +1,9 @@
-          def image_tag = git_branch
-          def slashPosition = git_branch.indexOf('/')
-          if (slashPosition > 0) {
-            git_tag = git_branch.substring(slashPosition + 1, git_branch.length())
-            git_branch = git_branch.substring(0, slashPosition)
-            if (git_branch == "release" || git_branch == "hotfix" || git_branch == "bugfix" || git_branch == "tickets") {
-              image_tag = git_tag
-            }
-          }
-          dockerImageName = dockerImageName + image_tag
-          echo "dockerImageName: ${dockerImageName}"
-          dockerImage = docker.build(dockerImageName, "-f docker/Dockerfile .")
-        }
-      }
-    }
-
-    stage("Push Docker image") {
-      when {
-        anyOf {
-          branch "main"
-          branch "develop"
-          branch "bugfix/*"
-          branch "hotfix/*"
-          branch "release/*"
-          branch "tickets/*"
-        }
-      }
-      steps {
-        script {
-          docker.withRegistry("", registryCredential) {
-            dockerImage.push()
-          }
-        }
-      }
-    }
-
-    stage("Run tests") {
-      when {
-        anyOf {
-          branch "main"
-          branch "develop"
-          branch "bugfix/*"
-          branch "hotfix/*"
-          branch "release/*"
-          branch "tickets/*"
-          branch "PR-*"
-        }
-      }
-      steps {
-        script {
-          sh "docker build --pull -f docker/Dockerfile-dev -t love-commander-test  ."
-          sh "docker run --env LSST_DDS_PARTITION_PREFIX=ci --env LSST_SITE=ci love-commander-test:latest docker/run-tests.sh"
-        }
-      }
-    }
-
-    stage("Deploy documentation") {
-      agent {
-        docker {
-          alwaysPull true
-          image 'lsstts/develop-env:develop'
-          args "--entrypoint=''"
-        }
-      }
-      when {
-        anyOf {
-          branch "main"
-          branch "develop"
-        }
-      }
-      steps {
-        script {
-          sh """
-            source ${env.SAL_SETUP_FILE}
-            # Create docs
-            cd ./docsrc
-            pip install -r requirements.txt
-            sh ./create_docs.sh
-            cd ..
-            # Upload docs
-            pip install ltd-conveyor
-            ltd upload --product love-commander --git-ref ${GIT_BRANCH} --dir ./docs
-          """
-        }
-      }
-    }
-
-    stage("Trigger develop deployment") {
-      when {
-        branch "develop"
-      }
-      steps {
-        build(job: '../LOVE-integration-tools/develop', wait: false)
-      }
-    }
-    
-    stage("Trigger main deployment") {
-      when {
-        branch "main"
-      }
-      steps {
-        build(job: '../LOVE-integration-tools/main', wait: false)
-      }
-    }
-  }
-}
+@Library('JenkinsShared')_
+DevelopPipeline(
+    name: "love-commander",
+    module_name: "love.commander",
+    idl_names: [],
+    build_all_idl: true,
+    has_doc_site: false,
+    extra_packages: []
```

Comment:
You may omit lines 5 (`idl_names`) and 8 (`extra_packages`) since they contain the default values. See
https://github.com/lsst-ts/ts_jenkins_shared_library/blob/develop/vars/DevelopPipeline.groovy#L5-L15
Also, line 6 (`build_all_idl`) is ignored because of line 5. See
https://github.com/lsst-ts/ts_jenkins_shared_library/blob/develop/vars/DevelopPipeline.groovy#L30-L35
and
https://github.com/lsst-ts/ts_jenkins_shared_library/blob/develop/vars/DevelopPipeline.groovy#L116-L118
Please remove all three for clarity.